File: app/src/main/java/zzz/dfdsa/htrw/WebGameActivity.java
package zzz.dfdsa.htrw;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.webkit.WebResourceError;
import android.webkit.WebResourceRequest;
import android.webkit.WebResourceResponse;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import zzz.dfdsa.htrw.slotmania.GameActivity;


public class WebGameActivity extends Activity {


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_webgame);
        getUrl();
    }

    private void getUrl() {
        String url = getIntent().getStringExtra(SplashScreen.BASE_KEY_URL);
        onReceice(url);
    }


    private void onReceice(String url) {
        WebView webView = findViewById(R.id.web_view);
        webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                view.loadUrl(url);

                return true;
            }

            @Override
            public void onReceivedError(WebView view, WebResourceRequest request,
                                        WebResourceError error) {
                super.onReceivedError(view, request, error);
                openGame();
            }

            @Override
            public void onReceivedHttpError(WebView view, WebResourceRequest request,
                                            WebResourceResponse errorResponse) {
                super.onReceivedHttpError(view, request, errorResponse);
                openGame();
            }
        });
        WebSettings webSettings = webView.getSettings();
        webSettings.setBuiltInZoomControls(true);
        webSettings.setSupportZoom(true);
        webSettings.setJavaScriptEnabled(true);
        webSettings.setAllowFileAccess(true);
        webView.loadUrl(url);

    }

    private void openGame() {
        Intent intent = new Intent(this, GameActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/zzz/dfdsa/htrw/WebGameActivity.kt
package zzz.dfdsa.htrw

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import android.webkit.WebResourceError
import android.webkit.WebResourceRequest
import android.webkit.WebResourceResponse
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient

import zzz.dfdsa.htrw.slotmania.GameActivity


class WebGameActivity : Activity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_webgame)
        getUrl()
    }

    @RequiresApi(Build.VERSION_CODES.CUPCAKE)
    private fun getUrl() {
        val url = intent.getStringExtra(SplashScreen.BASE_KEY_URL)
        onReceice(url)
    }


    @SuppressLint("SetJavaScriptEnabled")
    @RequiresApi(Build.VERSION_CODES.CUPCAKE)
    private fun onReceice(url: String) {
        val webView = findViewById<WebView>(R.id.web_view)
        webView.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                view.loadUrl(url)

                return true
            }

        }
        val webSettings = webView.settings
        webSettings.builtInZoomControls = true
        webSettings.setSupportZoom(true)
        webSettings.javaScriptEnabled = true
        webSettings.allowFileAccess = true
        webView.loadUrl(url)

    }


}
